controller/scheduler/swarm.py

```diff
@@ -0,0 +1,134 @@
+import re
+import time
+
+from django.conf import settings
+from docker import Client
+
+from .states import JobState
+
+
+MATCH = re.compile(
+    r'(?P<app>[a-z0-9-]+)_?(?P<version>v[0-9]+)?\.?(?P<c_type>[a-z-_]+)?.(?P<c_num>[0-9]+)')
+
+
+class SwarmClient(object):
+
+    def __init__(self, target, auth, options, pkey):
+        self.target = settings.SWARM_HOST
+        # single global connection
+        self.registry = settings.REGISTRY_HOST + ':' + settings.REGISTRY_PORT
+        self.docker_cli = Client("tcp://{}:2395".format(self.target),
+                                 timeout=1200, version='1.17')
+
+    def create(self, name, image, command='', template=None, **kwargs):
+        """Create a container"""
+        cimage = self.registry + '/' + image
+        affinity = "affinity:container!=~/{}*/".format(re.split(r'_v\d.', name)[0])
+        l = locals().copy()
+        l.update(re.match(MATCH, name).groupdict())
+        mem = kwargs.get('memory', {}).get(l['c_type'])
+        if mem:
+            mem = mem.lower()
+            if mem[-2:-1].isalpha() and mem[-1].isalpha():
+                mem = mem[:-1]
+        cpu = kwargs.get('cpu', {}).get(l['c_type'])
+        self.docker_cli.create_container(image=cimage, name=name,
+                                         command=command.encode('utf-8'), mem_limit=mem,
+                                         cpu_shares=cpu,
+                                         environment=[affinity],
+                                         host_config={'PublishAllPorts': True})
+
+    def start(self, name):
+        """
+        Start a container
+        """
+        self.docker_cli.start(name)
+
+    def stop(self, name):
+        """
+        Stop a container
+        """
+        self.docker_cli.stop(name)
+
+    def destroy(self, name):
+        """
+        Destroy a container
+        """
+        self.stop(name)
```

Comment:
Okay, sounds reasonable, but what if we wanted to try destroying a container that is currently stopped such as when something like #3293 rolls around? Should we attempt to stop, but if that fails then we just destroy? Basically a

```
try:
    self.stop()
except:
    pass
finally:
    self.docker_cli.remove_container(name)
```